dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosRegistry.java

```diff
@@ -638,6 +639,12 @@ private URL buildURL(URL consumerURL, Instance instance) {
         return new DubboServiceAddressURL(url.getUrlAddress(), url.getUrlParam(), consumerURL, null);
     }
 
+
+    private void setWeight(Map<String, String> metadata, double nacosWeight) {
+        long weight = Math.round(org.apache.dubbo.rpc.cluster.Constants.DEFAULT_WEIGHT * nacosWeight);
```

Comment:
1. skip if nacos weight has not changed
2. We should consider if weight has been set in instance

Reply:
1. There is no flag to judge whether the weight of Nacos is modified，nacos will pass in the default weight of 1.0
![image](https://user-images.githubusercontent.com/26109828/217228731-a757836a-eced-43e1-8c60-b3aded3e2080.png)
2. If it is not set in the example, Nacos sends 1.0 by default, and the calculated result according to the code logic is 100, which is consistent with the default weight of dubbo;
![image](https://user-images.githubusercontent.com/26109828/217229356-4f469991-8e6a-42fb-86fa-b76c4f6ab681.png)

Reply:
> If it is not set in the instance, Nacos sends 1.0 by default, and the calculated result according to the code logic is 100, which is consistent with the default weight of dubbo;

If dubbo has set `200` in url, then set nacos weight as `3`, the weight should be `600` as expected.

Reply:
> > If it is not set in the instance, Nacos sends 1.0 by default, and the calculated result according to the code logic is 100, which is consistent with the default weight of dubbo;
> 
> If dubbo has set `200` in url, then set nacos weight as `3`, the weight should be `600` as expected.

Yes, because the weight field type of nacos is double, and the weight field type of dubbo is int, all weights set in nacos are multiplied by the default weight of 100, so that two decimal places are reserved; this logic is consistent in all instances;

Reply:
`Math.round(org.apache.dubbo.rpc.cluster.Constants.DEFAULT_WEIGHT * nacosWeight)` will override the weight from Dubbo provider

Reply:
> `Math.round(org.apache.dubbo.rpc.cluster.Constants.DEFAULT_WEIGHT * nacosWeight)` will override the weight from Dubbo provider


Yes, I may be able to modify it to judge whether Nacos is the default weight 1, if it is, the weight of Dubbo will not be updated, if not, the weight of Dubbo will be updated；
It’s just that the default weight 1 of Nacos is maintained in `com.alibaba.nacos.api.naming.pojo.Instance#weight`， which is not a constant, so a new constant value needs to be added to maintain the default value of Nacos weight;（what do you think of this way）

Reply:
What I mean is that when the weight of nacos is not 1, it may conflict with the weight of dubbo.

Reply:
> What I mean is that when the weight of nacos is not 1, it may conflict with the weight of dubbo.

Or you can provide a variable to the user to specify the default weight of Nacos, but this feels heavy;
From my understanding, if Nacos is used, then the weight control should be controlled by Nacos;

Reply:
I feel that it is better to modify Nacos, so Nacos is needed to be compatible with Dubbo's weight configuration

Reply:
It would be better to intoduce a new param like `registry_weight`. Set it in NacosRegistry and NacosServiceDiscoveryRegistry and get it in `org.apache.dubbo.rpc.cluster.loadbalance.AbstractLoadBalance#getWeight`.

Reply:
Are you referring to org.apache.dubbo.registry.nacos.NacosRegistry#createInstance? I see that when creating an instance here, the weight of the dubbo configuration is not set. Here, the weight of the dubbo configuration can be registered on Nacos, and the weight issued by Nacos shall prevail when obtaining it. In this way, there will be no scenarios covered by Nacos.
![image](https://user-images.githubusercontent.com/26109828/219523718-26663581-00a7-4ac7-bfe3-8b86341404ee.png)

Reply:
The changes are as shown in the figure (default rounding for decimals)
![image](https://user-images.githubusercontent.com/26109828/219524146-b4897fff-bd35-45b2-8ff4-f8fe90c7cce1.png)

Reply:
> The changes are as shown in the figure (default rounding for double) ![image](https://user-images.githubusercontent.com/26109828/219524146-b4897fff-bd35-45b2-8ff4-f8fe90c7cce1.png)

This is an good idea. But I am afraid that it would not work in ServiceDiscovery model.

Reply:
> > The changes are as shown in the figure (default rounding for double) ![image](https://user-images.githubusercontent.com/26109828/219524146-b4897fff-bd35-45b2-8ff4-f8fe90c7cce1.png)
> 
> This is an good idea. But I am afraid that it would not work in ServiceDiscovery model.


I verified it, it works. When registering, set the weight of dubbo to Nacos; the weight issued by Nacos can be set to Dubbo;


![image](https://user-images.githubusercontent.com/26109828/220078320-7fcac8e3-6db8-4e1f-b51c-393501ffa34a.png)

Reply:
> > > The changes are as shown in the figure (default rounding for double) ![image](https://user-images.githubusercontent.com/26109828/219524146-b4897fff-bd35-45b2-8ff4-f8fe90c7cce1.png)
> > 
> > 
> > This is an good idea. But I am afraid that it would not work in ServiceDiscovery model.
> 
> I verified it, it works. When registering, set the weight of dubbo to Nacos; the weight issued by Nacos can be set to Dubbo;
> 
> ![image](https://user-images.githubusercontent.com/26109828/220078320-7fcac8e3-6db8-4e1f-b51c-393501ffa34a.png)

If different service has different weight, it may conflict.

Reply:
For example, for interface A, I start three instances and set three different weights respectively. The set weights are stored in the org.apache.dubbo.common.URL parameter. When reporting, the weight is obtained from the URL parameter, and the delivered weight Get it from com.alibaba.nacos.api.naming.pojo.Instance#weight;
————————
比如A接口，我启动三个实例，分别设置三个不同的权重，设置的权重存储在org.apache.dubbo.common.URL 参数中，上报的时候从URL参数中获取到权重，下发的权重的时候从com.alibaba.nacos.api.naming.pojo.Instance#weight 获取；

Reply:
Dubbo 的应用级服务发现是把多个服务（接口）聚合为一个应用级的大服务注册到注册中心的，因此如果每个接口权重不一样的话，大服务注册到注册中心的权重就无法确认了

Reply:
明白了，我新加的这个代码应该合并在2.7.x里面